src/HeaderSecurity.php

```diff
@@ -148,16 +148,16 @@ public static function assertValid(mixed $value): void
      */
     public static function assertValidName(mixed $name): void
     {
-        if (! is_string($name)) {
+        if (! is_string($name) && ! is_numeric($name)) {
```

Comment:
I would expect them to be `string` anyway: casting a `float` to `string` is not our business, and may lead to trouble down the road.

Changing the regex: sure.

Casting `float` to a `string` is IMO a problem.

Reply:
Only integer like values are affected. Negative and positive. Integers that start with 0 are not converted, except singular 0.

https://3v4l.org/ei8MB
```
array(6) {
  [1]=>
  string(0) ""
  [-1]=>
  string(0) ""
  [0]=>
  string(0) ""
  ["0.1"]=>
  string(0) ""
  ["0123"]=>
  string(0) ""
  ["00"]=>
  string(0) ""
}
```

Reply:
The problem is with casts that can also lead to a scientific notation 

Thinking https://stackoverflow.com/q/1471674/347063

Reply:
The problem here is as follows:

- `ServerRequestFactory` calls on `Lamins\Diactoros\marshalHeadersFromSapi()`.
- `Laminas\Diactoros\marshalHeadersFromSapi()` loops through the `$_SERVER` array, looking for keys that match one of:
    - `REDIRECT_`
    - `CONTENT_`
    - `HTTP_`
- Keys that are of interest are then added to an array of discovered headers and returned to the factory.
- The factory uses the discovered headers to construct a `ServerRequest` instance
- `ServerRequest` takes that list of headers and passes them to `HeaderSecurity` to validate the names and values.

And this is where the issue comes in, because _PHP converts keys that are strings containing only integers to integers_. This happens at the point that `marshalHeadersFromSapi()` returns the header list, so once the `ServerRequest` starts to inject them into itself, they then fail validations.

I'm fine with not casting float to string, and can make that change, but the integer values are the ones that this patch is really concerned about. If I make the change to remove support for floats, will this resolve your reservations, @Ocramius ?